library.ml

```diff
@@ -3,64 +3,77 @@ open Printf
 open String
 open List
 
-class enemy=object (self)
-  val mutable name:string=""
-  val mutable health:int=0
-  val mutable dmg:int=0
-  method set_health n=health <- n
-  method set_name n=name <- n
-  method set_dmg n=dmg <- n
+module Enemy = struct
+  type t =
+    { name: string ;
+      health: int ;
+      dmg: int ; }
 
-  method appear:unit=
-    printf "An enemy ";print_string [red] name;printf " appeared!\n";
-    self#battle
-  method battle:unit=
-    if health>0 then begin
-      print_string [red] name;printf " attacks you for ";
-      print_string [red] (string_of_int dmg);printf " damage\n";
+  let empty = { name = "" ; health = 0 ; dmg = 0 ; }
```

Comment:
Since creating a new instance of the `t` type requires us to "bind" (a.k.a assign) all the fields in the record, it is common to define a default instance of it that you can then copy (and override) from. Again, there's nothing magical about the name `empty`, it's just the customary name :)